File: ClientApp/src/app/data/models/Trip.ts
import { SafeUrl } from "@angular/platform-browser";
import { TripImage } from "./TripImage";

export class Trip{

    constructor(
        public id: string,
        public name: string,
        public description: string,
        public price: number,
        public imagePath: SafeUrl,
        public tripImageId: string,
        public tripImage: TripImage,
        public distance?: number,
        public duration?: number
    ){}
}

File: Models/Trip.cs
using System;
using System.ComponentModel.DataAnnotations;

public class Trip{

    public Guid Id { get; set; }

    [Required]
    [MaxLength(50)]
    public string Name { get; set; }

    [Required]
    [MaxLength(250)]
    public string Description { get; set; }
    
    [Required]
    public int Price { get; set; }

    [Required]
    public string ImagePath { get; set; }

    [Required]
    public Guid TripImageId { get; set; }

    public TripImage TripImage { get; set; }

    public int Distance { get; set; }
    
    public int Duration { get; set; }
}
